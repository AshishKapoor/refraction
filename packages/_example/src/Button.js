// @flow
import styled from '@emotion/styled';

const Button = styled.button`
  color: red;
`;

// @component
export default Button;
