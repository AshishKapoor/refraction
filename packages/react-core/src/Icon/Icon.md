With the `Icon` component you can insert one of the supported icons
in your component:

```js
<Icon name="globe" style={{ fontSize: '2em' }} />
```

This is the complete list of available icons:

```js noeditor
const iconNames = require('./icons/quid-icons.json').icons.map(
  ({ icon }) => icon.tags[0]
);
const cell = {
  maxWidth: 0,
};
const icon = {
  padding: '.5em 1em',
  fontSize: '2em',
};
const name = {
  whiteSpace: 'normal',
  wordWrap: 'break-word',
  minWidth: 0,
  marginLeft: '1em',
};

<div style={{ overflow: 'auto', maxHeight: 400 }}>
  <table className="DemoTable">
    <tbody>
      {iconNames
        .reduce((acc, icon, i) => {
          i % 2 === 1 &&
            acc.push([
              icon.split(',')[0],
              iconNames[i + 1] && iconNames[i + 1].split(',')[0],
            ]);
          return acc;
        }, [])
        .map((icons, i) => (
          <tr key={i}>
            <td style={cell}>
              <div style={{ display: 'flex', alignItems: 'center' }}>
                <Icon style={icon} name={icons[0]} />
                <div style={name}>{icons[0]}</div>
              </div>
            </td>

            <td style={cell}>
              <div style={{ display: 'flex', alignItems: 'center' }}>
                <Icon style={icon} name={icons[1]} />
                <div style={name}>{icons[1]}</div>
              </div>
            </td>
          </tr>
        ))}
    </tbody>
  </table>
</div>;
```
