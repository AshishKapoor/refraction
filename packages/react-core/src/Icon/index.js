// @flow
import * as React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled/macro';

type Props = {
  /** Icon name */
  name: string,
  /** Forces the icon to be rendered with a squared area, this helps icons alignment */
  squared?: boolean,
};

const Icon = styled(({ name, squared = true, ...props }: Props) => (
  <i {...props}>{name}</i>
))`
  /* use !important to prevent issues with browser extensions that change fonts */
  font-family: 'quid-icons' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: inherit;

  /* Enable Ligatures */
  letter-spacing: 0;
  font-feature-settings: 'liga';
  font-variant-ligatures: discretionary-ligatures;

  /* Better Font Rendering */
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;

  ${props =>
    props.squared &&
    css`
      display: inline-flex;
      align-items: center;
      justify-content: center;
      height: 1em;
      width: 1em;
    `};
`;

// @component
export default Icon;
