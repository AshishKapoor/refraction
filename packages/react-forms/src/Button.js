// @flow
export { Button as default } from '@quid/react-core';
