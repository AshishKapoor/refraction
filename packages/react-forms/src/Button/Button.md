Alias for the [`@quid/react-core#Button`](#button) component.

It is exposed from this package for convenience.
