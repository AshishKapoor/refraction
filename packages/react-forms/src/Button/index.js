// @flow
import React, { type Node } from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { Link } from 'react-router-dom';
import Color from 'color';
import { withFallback as wf, textStyles } from '@quid/theme';

const OKAY = '#039849';
const WARNING = '#FFCE03';
const HAZARD = '#E61E27';

type Props = {
  /** The "color" of the button */
  importance?: 'primary' | 'secondary' | 'okay' | 'warning' | 'hazard',
  /** Make the button background transparent */
  transparent?: boolean,
  /** Button height */
  size?: 'regular' | 'small',
  /** What to render inside the button */
  children?: Node,
  /** Normal `href` URL, if present, the button will render as anchor (`<a />`) */
  href?: string,
  /** react-router URL, if present, the button will render as `Link` */
  to?: string | Object,
  /** when `true`, the button can't be clicked */
  disabled?: boolean,
};

const hoverMod = color =>
  Color(color)
    .darken(0.1)
    .string();
const activeMod = color =>
  Color(color)
    .darken(0.2)
    .string();

const reset = css`
  border: 0;
  padding: 0;
  background: none;
  cursor: pointer;
  font-size: inherit;
  font-family: inherit;
  text-decoration: none;
  box-sizing: border-box;
  color: inherit;
`;

const BaseButton = styled.button`
  ${reset}

  ${textStyles('normal', 'bold')}
  display: inline-block;
  border-radius: 2px;
  line-height: ${props => (props.size === 'small' ? 1.9 : 2.3)};
  padding-left: 0.769em;
  padding-right: 0.769em;
  transition: padding 0.2s ease-in-out, background 0.2s ease-in-out;
  color: ${wf(props => props.theme.primary)};

  ${wf(props => {
    const variations = {
      primary: props.theme.colors.aqua,
      secondary: props.theme.colors.gray2,
      okay: OKAY,
      warning: WARNING,
      hazard: HAZARD,
    };

    const color = variations[props.importance];

    return css`
      background-color: ${color};
      &:hover {
        background-color: ${hoverMod(color)};
      }
      &:active {
        background-color: ${activeMod(color)};
      }
      &:focus:not(:focus-visible) {
        outline: 0;
      }
      &:focus-visible {
        outline: 0;
        box-shadow: 0 0 0 0.5px white, 0 0 2px 2px ${color};
      }
    `;
  })}

  ${props =>
    props.disabled &&
    css`
      cursor: default;
      &,
      &:hover,
      &:focus,
      &:active {
        opacity: 0.4;
      }
    `}
  ${props =>
    props.disabled &&
    props.transparent &&
    css`
      background-color: transparent;
    `}
`;

const Button = (props: Props) => {
  let tag;
  if (props.to && !props.disabled) {
    tag = Link;
  } else if (props.href || (props.to && props.disabled)) {
    tag = 'a';
  } else {
    tag = 'button';
  }

  return <BaseButton as={tag} {...props} />;
};
Button.defaultProps = {
  importance: 'primary',
};

export default Button;
