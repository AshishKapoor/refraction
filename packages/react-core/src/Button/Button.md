The `Button` component is used to let the user perform an action
once it's clicked:

```js
<Button importance="primary">I'm a button</Button>
```

These are all the supported style variations:

```js
import Icon from '../Icon';

<table className="DemoTable">
  <thead>
    <tr>
      <th />
      <th>Default</th>
      <th>Disabled</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Primary</td>
      <td>
        <Button importance="primary">Button</Button>
      </td>
      <td>
        <Button importance="primary" disabled>
          Button
        </Button>
      </td>
    </tr>
    <tr>
      <td>Primary Small</td>
      <td>
        <Button importance="primary" size="small">
          Button
        </Button>
      </td>
      <td>
        <Button importance="primary" size="small" disabled>
          Button
        </Button>
      </td>
    </tr>
    <tr>
      <td>Primary with Icon</td>
      <td>
        <Button importance="primary">
          <Icon name="scatter" />
        </Button>
      </td>
      <td>
        <Button importance="primary" disabled>
          <Icon name="scatter" />
        </Button>
      </td>
    </tr>
    <tr>
      <td>Primary Small with Icon</td>
      <td>
        <Button importance="primary" size="small">
          <Icon name="scatter" />
        </Button>
      </td>
      <td>
        <Button importance="primary" size="small" disabled>
          <Icon name="scatter" />
        </Button>
      </td>
    </tr>
    <tr>
      <td>Primary with both</td>
      <td>
        <Button importance="primary">
          <Icon name="download" /> Button
        </Button>
      </td>
      <td>
        <Button importance="primary" disabled>
          <Icon name="download" /> Button
        </Button>
      </td>
    </tr>
    <tr>
      <td>Primary with both (alt)</td>
      <td>
        <Button importance="primary">
          Button
          <Icon name="download" />
        </Button>
      </td>
      <td>
        <Button importance="primary" disabled>
          Button
          <Icon name="download" />
        </Button>
      </td>
    </tr>
    <tr>
      <td>Primary Small with both</td>
      <td>
        <Button importance="primary" size="small">
          <Icon name="download" /> Button
        </Button>
      </td>
      <td>
        <Button importance="primary" size="small" disabled>
          <Icon name="download" /> Button
        </Button>
      </td>
    </tr>
    <tr>
      <td colSpan="3">
        <hr />
      </td>
    </tr>
    <tr>
      <td>Secondary</td>
      <td>
        <Button importance="secondary">Button</Button>
      </td>
      <td>
        <Button importance="secondary" disabled>
          Button
        </Button>
      </td>
    </tr>
    <tr>
      <td>Secondary Small</td>
      <td>
        <Button importance="secondary" size="small">
          Button
        </Button>
      </td>
      <td>
        <Button importance="secondary" size="small" disabled>
          Button
        </Button>
      </td>
    </tr>
    <tr>
      <td>Secondary with Icon</td>
      <td>
        <Button importance="secondary">
          <Icon name="scatter" />
        </Button>
      </td>
      <td>
        <Button importance="secondary" disabled>
          <Icon name="scatter" />
        </Button>
      </td>
    </tr>
    <tr>
      <td>Secondary Small with Icon</td>
      <td>
        <Button importance="secondary" size="small">
          <Icon name="scatter" />
        </Button>
      </td>
      <td>
        <Button importance="secondary" size="small" disabled>
          <Icon name="scatter" />
        </Button>
      </td>
    </tr>
    <tr>
      <td>Secondary with both</td>
      <td>
        <Button importance="secondary">
          <Icon name="download" /> Button
        </Button>
      </td>
      <td>
        <Button importance="secondary" disabled>
          <Icon name="download" /> Button
        </Button>
      </td>
    </tr>
    <tr>
      <td>Secondary Small with both</td>
      <td>
        <Button importance="secondary" size="small">
          <Icon name="download" /> Button
        </Button>
      </td>
      <td>
        <Button importance="secondary" size="small" disabled>
          <Icon name="download" /> Button
        </Button>
      </td>
    </tr>
    <tr>
      <td colSpan="3">
        <hr />
      </td>
    </tr>
    <tr>
      <td>Okay</td>
      <td>
        <Button importance="okay">Button</Button>
      </td>
      <td>
        <Button importance="okay" disabled>
          Button
        </Button>
      </td>
    </tr>
    <tr>
      <td>Warning</td>
      <td>
        <Button importance="warning">Button</Button>
      </td>
      <td>
        <Button importance="warning" disabled>
          Button
        </Button>
      </td>
    </tr>
    <tr>
      <td>Hazard</td>
      <td>
        <Button importance="hazard">Button</Button>
      </td>
      <td>
        <Button importance="hazard" disabled>
          Button
        </Button>
      </td>
    </tr>
  </tbody>
</table>;
```
