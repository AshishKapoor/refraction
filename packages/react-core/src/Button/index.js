// @flow
import * as React from 'react';
import css from '@emotion/css/macro';
import styled from '@emotion/styled/macro';
import { Link } from 'react-router-dom';
import Color from 'color';
import { withFallback as wf, textStyles } from '@quid/theme';
import Icon from '../Icon';

const OKAY = '#039849';
const WARNING = '#FFCE03';
const HAZARD = '#E61E27';

export type Importance =
  | 'primary'
  | 'secondary'
  | 'okay'
  | 'warning'
  | 'hazard';

export type Size = 'regular' | 'small';

type Props = {
  /** The "color" of the button */
  importance?: Importance,
  /** Make the button background transparent */
  transparent?: boolean,
  /** Button height */
  size?: Size,
  /** What to render inside the button */
  children?: React.Node,
  /** Normal `href` URL, if present, the button will render as anchor (`<a />`) */
  href?: string,
  /** react-router URL, if present, the button will render as `Link` */
  to?: string | Object,
  /** when `true`, the button can't be clicked */
  disabled?: boolean,
};

const hoverMod = color =>
  Color(color)
    .darken(0.1)
    .string();
const activeMod = color =>
  Color(color)
    .darken(0.2)
    .string();

const reset = css`
  border: 0;
  padding: 0;
  background: none;
  cursor: pointer;
  font-size: inherit;
  font-family: inherit;
  text-decoration: none;
  box-sizing: border-box;
  color: inherit;
`;

const Button = styled(
  ({ to, href, transparent, size, importance, disabled, ...props }: Props) => {
    let Tag, specificProps;
    if (to && !disabled) {
      Tag = Link;
      specificProps = { to };
    } else if ((href || to) && !disabled) {
      Tag = 'a';
      specificProps = { href };
    } else {
      Tag = 'button';
      specificProps = { disabled };
    }

    return <Tag {...specificProps} {...props} />;
  }
)`
  ${reset}

  ${textStyles('normal', 'bold')}
  display: inline-block;
  border-radius: 2px;
  line-height: ${props => (props.size === 'small' ? 1.9 : 2.3)};
  padding-left: 0.769em;
  padding-right: 0.769em;
  transition: padding 0.2s ease-in-out, background 0.2s ease-in-out;
  color: ${wf(props => props.theme.colors.black)};

  ${wf(props => {
    const variations = {
      primary: props.theme.colors.aqua,
      secondary: props.theme.colors.gray2,
      okay: OKAY,
      warning: WARNING,
      hazard: HAZARD,
    };

    const color = variations[props.importance];

    return css`
      background-color: ${color};
      &:hover {
        background-color: ${hoverMod(color)};
      }
      &:active {
        background-color: ${activeMod(color)};
      }
      &:focus:not(:focus-visible) {
        outline: 0;
      }
      &:focus-visible {
        outline: 0;
        box-shadow: 0 0 0 0.5px white, 0 0 2px 2px ${color};
      }
    `;
  })}

  ${props =>
    props.disabled &&
    css`
      cursor: default;
      &,
      &:hover,
      &:focus,
      &:active {
        opacity: 0.4;
      }
    `}
  ${props =>
    props.disabled &&
    props.transparent &&
    css`
      background-color: transparent;
    `}

  ${Icon} {
      display: block;
      line-height: 1;
      font-size: 1.42em;
      margin-left: -6px;
      margin-right: -6px;
      margin-top: 5px;
      margin-bottom: 5px;

      ${props =>
        props.size === 'small' &&
        css`
          font-size: 1em;
        `}

      ${props =>
        React.Children.count(props.children) > 1 &&
        css`
          display: inline-block;
          position: relative;
          font-size: 1em;
          margin-left: 0;
          bottom: -1px;
          margin-right: 0.35em;
        `}
  }
`;

Button.defaultProps = {
  importance: 'primary',
  size: 'regular',
};

// @component
export default Button;
