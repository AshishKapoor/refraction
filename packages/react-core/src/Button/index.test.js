// @flow
import * as React from 'react';
import { mount } from 'enzyme';
import Button from '.';
import { Link } from 'react-router-dom';
import Icon from '../Icon';

jest.mock('react-router-dom');

it('renders the expected markup', () => {
  const wrapper = mount(<Button data-action="foo">Hello, World!</Button>);
  expect(wrapper).toMatchSnapshot();
});

it('renders a Button with link', () => {
  const wrapper = mount(
    <Button data-action="foo" to="/">
      Hello, World!
    </Button>
  );
  expect(wrapper.find(Link)).toBeDefined();
});

it('renders a Button with small size', () => {
  const wrapper = mount(<Button size="small">Hello, World!</Button>);
  expect(wrapper).toMatchSnapshot();
});

it('renders a disabled and transparent Button', () => {
  const wrapper = mount(
    <Button transparent disabled>
      Hello, World!
    </Button>
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders a Button with external link', () => {
  const wrapper = mount(
    <Button data-action="foo" href="https://example.org">
      Hello, World!
    </Button>
  );
  expect(wrapper.find('a[href]').props().href).toBe('https://example.org');
});

it('renders a disabled Button with external link', () => {
  const wrapper = mount(
    <Button data-action="foo" disabled href="https://example.org">
      Hello, World!
    </Button>
  );
  expect(wrapper.find('a[href]')).toHaveLength(0);
});

it('renders a link without href if `to` is given but `disabled={true}`', () => {
  const wrapper = mount(
    <Button data-action="foo" to="/" disabled={true}>
      Hello, World!
    </Button>
  );
  expect(wrapper.find(Link)).toHaveLength(0);
});

it('applies Button classes to child if child is Icon', () => {
  const wrapper = mount(
    <Button data-action="foo">
      <Icon name="arrow" />
    </Button>
  );
  expect(wrapper.find('i').props().className).toMatchInlineSnapshot(
    `"css-1k94zxf-Icon e1t5eso00"`
  );
});

it('does not apply Button classes to child if child is not Icon', () => {
  const wrapper = mount(
    <Button data-action="foo">
      <div data-target>Hello, World!</div>
    </Button>
  );
  expect(wrapper.find('[data-target]').props().className).toMatchInlineSnapshot(
    `undefined`
  );
});

it('applies Button classes to child if child is Icon with near text', () => {
  const wrapper = mount(
    <Button>
      <Icon name="arrow" />
      Foobar
    </Button>
  );
  expect(wrapper.find('i').props().className).toMatchInlineSnapshot(
    `"css-1k94zxf-Icon e1t5eso00"`
  );
});

// it('renders properly with isGroupChild property', () => {
//   const wrapper = mount(<Button isGroupChild />);
//   expect(wrapper).toMatchSnapshot();
// });
