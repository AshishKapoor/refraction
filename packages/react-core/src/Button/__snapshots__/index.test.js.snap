// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Button with small size 1`] = `
.emotion-0 {
  border: 0;
  padding: 0;
  background: none;
  cursor: pointer;
  font-size: inherit;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  color: inherit;
  font-family: IBM Plex Sans,Lucida Grande,Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.57;
  font-weight: bold;
  display: inline-block;
  border-radius: 2px;
  height: 24px;
  padding-left: 0.769em;
  padding-right: 0.769em;
  -webkit-transition: padding 0.2s ease-in-out,background 0.2s ease-in-out;
  transition: padding 0.2s ease-in-out,background 0.2s ease-in-out;
  color: #121212;
  border: 1px solid transparent;
  background-color: #1ED7D1;
}

.emotion-0:hover,
.emotion-0[data-state*='hover'] {
  background-color: hsl(178.10000000000002,75.5%,43.2%);
}

.emotion-0:active,
.emotion-0[data-state*='active'] {
  background-color: hsl(178.10000000000002,75.5%,38.4%);
}

.emotion-0:focus:not(:focus-visible) {
  outline: 0;
}

.emotion-0:focus-visible,
.emotion-0[data-state*='focus'] {
  outline: 0;
  box-shadow: 0 0 0 0.5px #FFFFFF,0 0 2px 2px #1ED7D1;
}

.emotion-0 .e1t5eso00 {
  display: block;
  line-height: 1;
  font-size: 1.42em;
  margin-left: -6px;
  margin-right: -6px;
  margin-top: 5px;
  margin-bottom: 5px;
  font-size: 1em;
}

<Button
  importance="primary"
  size="small"
>
  <button
    className="emotion-0 emotion-1"
  >
    Hello, World!
  </button>
</Button>
`;

exports[`renders a disabled and transparent Button 1`] = `
.emotion-0 {
  border: 0;
  padding: 0;
  background: none;
  cursor: pointer;
  font-size: inherit;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  color: inherit;
  font-family: IBM Plex Sans,Lucida Grande,Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.57;
  font-weight: bold;
  display: inline-block;
  border-radius: 2px;
  height: 32px;
  padding-left: 0.769em;
  padding-right: 0.769em;
  -webkit-transition: padding 0.2s ease-in-out,background 0.2s ease-in-out;
  transition: padding 0.2s ease-in-out,background 0.2s ease-in-out;
  color: #121212;
  border: 1px solid transparent;
  background-color: #1ED7D1;
  cursor: default;
  background-color: transparent;
}

.emotion-0:hover,
.emotion-0[data-state*='hover'] {
  background-color: hsl(178.10000000000002,75.5%,43.2%);
}

.emotion-0:active,
.emotion-0[data-state*='active'] {
  background-color: hsl(178.10000000000002,75.5%,38.4%);
}

.emotion-0:focus:not(:focus-visible) {
  outline: 0;
}

.emotion-0:focus-visible,
.emotion-0[data-state*='focus'] {
  outline: 0;
  box-shadow: 0 0 0 0.5px #FFFFFF,0 0 2px 2px #1ED7D1;
}

.emotion-0,
.emotion-0:hover,
.emotion-0:focus,
.emotion-0:active {
  opacity: 0.4;
}

.emotion-0 .e1t5eso00 {
  display: block;
  line-height: 1;
  font-size: 1.42em;
  margin-left: -6px;
  margin-right: -6px;
  margin-top: 5px;
  margin-bottom: 5px;
}

<Button
  disabled={true}
  importance="primary"
  size="regular"
  transparent={true}
>
  <button
    className="emotion-0 emotion-1"
    disabled={true}
  >
    Hello, World!
  </button>
</Button>
`;

exports[`renders the expected markup 1`] = `
.emotion-0 {
  border: 0;
  padding: 0;
  background: none;
  cursor: pointer;
  font-size: inherit;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  color: inherit;
  font-family: IBM Plex Sans,Lucida Grande,Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.57;
  font-weight: bold;
  display: inline-block;
  border-radius: 2px;
  height: 32px;
  padding-left: 0.769em;
  padding-right: 0.769em;
  -webkit-transition: padding 0.2s ease-in-out,background 0.2s ease-in-out;
  transition: padding 0.2s ease-in-out,background 0.2s ease-in-out;
  color: #121212;
  border: 1px solid transparent;
  background-color: #1ED7D1;
}

.emotion-0:hover,
.emotion-0[data-state*='hover'] {
  background-color: hsl(178.10000000000002,75.5%,43.2%);
}

.emotion-0:active,
.emotion-0[data-state*='active'] {
  background-color: hsl(178.10000000000002,75.5%,38.4%);
}

.emotion-0:focus:not(:focus-visible) {
  outline: 0;
}

.emotion-0:focus-visible,
.emotion-0[data-state*='focus'] {
  outline: 0;
  box-shadow: 0 0 0 0.5px #FFFFFF,0 0 2px 2px #1ED7D1;
}

.emotion-0 .e1t5eso00 {
  display: block;
  line-height: 1;
  font-size: 1.42em;
  margin-left: -6px;
  margin-right: -6px;
  margin-top: 5px;
  margin-bottom: 5px;
}

<Button
  data-action="foo"
  importance="primary"
  size="regular"
>
  <button
    className="emotion-0 emotion-1"
    data-action="foo"
  >
    Hello, World!
  </button>
</Button>
`;
